xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-wikimacro/xwiki-platform-rendering-wikimacro-api/src/main/java/org/xwiki/rendering/macro/wikibridge/WikiMacroParameterDescriptor.java

```diff
@@ -60,6 +79,11 @@
      */
     private Object defaultValue;
 
+    /**
+     * Type of the parameter.
+     */
+    private String type;
```

Comment:
This should be a Type, it's not WikiMacroParameterDescriptor job to do String -> Type, it should be done when doing xobject -> WikiMacroParameterDescriptor. Also forget about that very limited mapping, the value in the xobject should be the actual serialized Type ("jav.util.List<Integer>"), you can use org.xwiki.component.util.ReflectionUtils to parse it.

Also your implementation of getType() is wrong, looks as DefaultPropertyDescriptor#getPropertyClass() (except that the default would be String instead of null probably).

Reply:
Thanks @tmortagne, I agree with what you say. I just don't understand why the value of the xobject is `java.util.List<Integer>`, isn't it `String`? Also if I don't use a mapping, how can I provide the possible values in the Document initializer?